base/hash.h

```diff
@@ -33,7 +38,98 @@ inline uint32_t Fingerprint32(const std::string& str) {
   return uint32_t(res >> 32) ^ uint32_t(res);
 }
 
+namespace detail {
+
+// Should use std::has_unique_object_representations but we do not have in C++14.
+template<typename Hasher, typename T>
+std::enable_if_t<std::is_arithmetic<T>::value || std::is_enum<T>::value>
```

Comment:
Pls restrict to is_integral, handling double types correctly would require more work.